### ISSUE TITLE
Support for CMake

### ISSUE THREAD

**Issue by henryborchers** [mipops/dvrescue#17]
CMake is a meta-build system that generates many different build systems, including GNU Makefiles, Visual Studio projects, Ninja, Xcode projects. CMake has become the de facto standard for C and C++ projects because it much easier to build across the various platforms.

It also includes built-in support for packaging tools. (such as Wix Toolset aka .msi files , Nullsoft installers, DNG, RPM, Debian .deb files.

If you're looking to make this software survive for a long time, there is nothing better out that for building with many toolchains.

**Comment by henryborchers** [comment]
@JeromeMartinez, if you need any help. CMake is something that I've spent a lot of time with.

**Comment by JeromeMartinez** [comment]
@henryborchers project dependencies [already have CMake project files](https://github.com/MediaArea/MediaInfoLib/tree/master/Project/CMake), we'll provide a basic `CMakeList.txt` for this project and it will be up to you to improve it.

> It also includes built-in support for packaging tools. (such as Wix Toolset aka .msi files , Nullsoft installers, DNG, RPM, Debian .deb files.

One important "feature" I like with autotools is the ability to provide a "source" (not exactly source as we have to build some files) package without too much extra size (without providing autotools completely) and which can be run on any Linux or macOS machine with just a compiler installed, CMake is not installed as often as a compiler. Last time I checked CMake, this was not possible without providing the whole CMake package (too big), did it change?

> If you're looking to make this software survive for a long time, there is nothing better out that for building with many toolchains.

You have some experience with lot of platforms and use to provide binaries for several "stores", so we have had to deal with many toolchains...
